oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/userinfo/DefaultReactiveOAuth2UserService.java

```diff
@@ -108,7 +108,7 @@
 					authenticationMethod);
 			// @formatter:off
 			Mono<Map<String, Object>> userAttributes = requestHeadersSpec.retrieve()
-					.onStatus((s) -> s != HttpStatus.OK, (response) ->
+					.onStatus((s) -> !s.is2xxSuccessful(), (response) ->
```

Comment:
It doesn't make sense to accept all 2xx status code. For example, `203 Non-Authoritative Information` or `205 Reset Content`.

Reply:
@BenjaminFaal I reviewed `DefaultOAuth2UserService`. The associated `RestTemplate` uses `DefaultResponseErrorHandler` (indirectly), which checks for errors using `HttpStatus.isError()`.

Therefore, we can align `DefaultReactiveOAuth2UserService` with `DefaultOAuth2UserService` by applying the change `.onStatus(HttpStatus::isError, (response) -> ...`

Can you please also add a test for your use case, where the status returned is 201.

Reply:
@jgrandja done, see c4f6f3f3d8b6437f9de331c07b70761a08db4196